### PR TITLE
Refactor linkify

### PR DESCRIPTION
**Pull request by nqhhdev**

### CODE REVIEW

File: test/linkify_text_test.dart [hunk 1]
```diff
@@ -1 +1,406 @@
...
+        }
+      }
+    });
+
+    test('Phone codes for Zones 3–4 countries (Europe)', () {
+      const phoneEntries = [
+        'Greece: +30 210-123-4567',
+        'Netherlands: +31 612-345-6789',
+        'Belgium: +32 2-345-6789',
+        'France: +33 1-234-567-890',
```
Comment (hoangdat) [inline test/linkify_text_test.dart:254]: - [x] should add more variety cases for: Fr, Ru, 
- [x] should add test case for Asia zone

File: lib/src/linkify.dart [hunk 2]
```diff
@@ -43,7 +46,7 @@ class LinkifyText extends StatelessWidget {
   ///       print("${link.value} hashtag was tapped");
   ///   });
   /// ```
-  final void Function(Link)? onTap;
+  final void Function(TapDownDetails, Link)? onLinkTap;
```
Comment (dab246) [inline lib/src/linkify.dart]: We should not delete `onTap` function. Should create new `onTapDown ` function.

File: lib/src/linkify.dart [hunk 3]
```diff
@@ -322,21 +331,25 @@ class LinkifySelectableText extends StatelessWidget {
...
```
Comment (dab246) [inline lib/src/linkify.dart]: idem

**Review by dab246 — COMMENTED** [review]
IMO, we just need to change the necessary things to be able to merge with the upstream.

**Pull request merged**